pol-core/bscript/compiler/ast/Identifier.h

```diff
@@ -12,14 +13,20 @@ class Variable;
 class Identifier : public Expression
 {
 public:
-  Identifier( const SourceLocation&, std::string scope, std::string name );
+  Identifier( const SourceLocation&, const ScopableName& scoped_name );
+  Identifier( const SourceLocation&, const std::string& name );
 
   void accept( NodeVisitor& ) override;
   void describe_to( std::string& ) const override;
 
-  const std::string scope;
-  const std::string name;
+  // Returns foo, Animal::foo but never ::foo
+  std::string string() const;
```

Comment:
Why this method? name() would return const ref, that would be enough since it can be captured as copy or reference

Reply:
Never mind, this goes down to ScopeName. Where it's a dynamic build string